sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py

```diff
@@ -791,13 +803,62 @@ def __init__(self, name: str, description: str, spec: Any, auth: OpenApiAuthDeta
     @property
     def definitions(self) -> List[OpenApiToolDefinition]:
         """
-        Get the OpenApi tool definitions.
+        Get the list of all API definitions for the tool.
 
-        :return: A list of tool definitions.
+        :return: A list of OpenAPI tool definitions.
         :rtype: List[ToolDefinition]
         """
         return self._definitions
 
+    def add_definition(
+        self, 
+        name: str, 
+        description: str, 
+        spec: Any, 
+        auth: Optional[OpenApiAuthDetails] = None
+    ) -> None:
+        """
+        Adds a new API definition dynamically.
+        Raises a ValueError if a definition with the same name already exists.
+
+        :param name: The name of the API.
+        :param description: The description of the API.
+        :param spec: The API specification.
+        :param auth: Optional authentication details for this particular API definition.
+                     If not provided, the tool's default authentication details will be used.
+        :raises ValueError: If a definition with the same name exists.
+        """
+        # Check if a definition with the same name exists.
+        if any(defn.openapi.name == name for defn in self._definitions):
+            raise ValueError(f"Definition '{name}' already exists and cannot be added again.")
+        
+        # Use provided auth if specified, otherwise use default
+        auth_to_use = auth if auth is not None else self._default_auth
+        
+        new_definition = OpenApiToolDefinition(
+            openapi=OpenApiFunctionDefinition(
+                name=name, 
+                description=description, 
+                spec=spec, 
+                auth=auth_to_use
+            )
+        )
+        self._definitions.append(new_definition)
+
+    def remove_definition(self, name: str) -> None:
+        """
+        Removes an API definition based on its name.
+
+        :param name: The name of the API definition to remove.
+        :raises ValueError: If the definition with the specified name does not exist.
+        """
+        for definition in self._definitions:
```

Comment:
Should we make `_definitions` a dictionary, in this case we will ensure that we have only one definition with given name?

Reply:
Let's check that for the next PR. Will discuss with you about that.